### DISCUSSION TITLE
Checking for non-empty values

### DISCUSSION THREAD

**Discussion by viveleroi** [gvergnaud/ts-pattern#150] [Q&A]
What's the recommended way to ensure a value is a string and *not empty*:

I'm matching an object with structure `{ text: string }` but I need to ensure the `text` string is also not empty. If it's empty, I still want the `otherwise` method to trigger.

```js
const text = match(value)
    .with({ text: P.string }, selected => selected.text)
    .otherwise(() => placeholder)
```

So far, it seems like this is what I need to do, I don't see any shorter ways:

```js
function isNonEmptyString(value: unknown): value is string {
  return typeof value === 'string' && value !== ''
}

const text = match(value)
  .with(
    {
      text: P.when(isNonEmptyString)
    },
    selected => selected.text
  )
  .otherwise(() => placeholder)
```

**Comment by gvergnaud** [comment]
Sorry for the late reply, but with TS-Pattern v5, you can now use `P.string.minLength(1)`!

```ts
const text = match(value)
  .with(
    { text: P.string.minLength(1) },
    selected => selected.text
  )
  .otherwise(() => placeholder)
```
